docs/csharp/snippets/discards/discard-pattern2.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+
+public class DiscardPatternMatching
+{
+    public static void DiscardSwitchExample()
+    {
+        // <DiscardSwitchExample>
+        object[] objects = { CultureInfo.CurrentCulture,
+                           CultureInfo.CurrentCulture.DateTimeFormat,
+                           CultureInfo.CurrentCulture.NumberFormat,
+                           new ArgumentException(), null };
+        foreach (var obj in objects)
+            ProvidesFormatInfo(obj);
+
+        static void ProvidesFormatInfo(object obj)
+        {
+            switch (obj)
+            {
+                case IFormatProvider fmt:
+                    Console.WriteLine($"{fmt} object");
+                    break;
+                case null:
+                    Console.Write("A null object reference: ");
+                    Console.WriteLine("Its use could result in a NullReferenceException");
+                    break;
+                case object _:
+                    Console.WriteLine("Some object type without format information");
+                    break;
+            }
+        }
+        // The example displays the following output:
+        //    en-US object
+        //    System.Globalization.DateTimeFormatInfo object
+        //    System.Globalization.NumberFormatInfo object
+        //    Some object type without format information
+        //    A null object reference: Its use could result in a NullReferenceException
```

Comment:
```suggestion
        static void ProvidesFormatInfo(object obj) =>
            Console.WriteLine(obj switch
            {
                IFormatProvider fmt => $"{fmt.GetType()} object",
                null => "A null object reference: Its use could result in a NullReferenceException",
                _ => "Some object type without format information"
            });
        // The example displays the following output:
        //    System.Globalization.CultureInfo object
        //    System.Globalization.DateTimeFormatInfo object
        //    System.Globalization.NumberFormatInfo object
        //    Some object type without format information
        //    A null object reference: Its use could result in a NullReferenceException
```

Not sure if this is better, but when I saw this it made me think of a switch expression use case. Also, the example output doesn't seem correct. [See here](https://dotnetfiddle.net/h7GTEj).

Reply:
Hmm. Output matches for me.